hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/EdgeAPI.java

```diff
@@ -157,6 +159,106 @@ public String create(@Context GraphManager manager,
         });
     }
 
+    /**
+     * TODO: PUT method should support list in huge-common, use POST first..
+     * TODO: Adapter for createIfNotExist
+     * Batch update steps are same like vertices
+     **/
+    @POST
+    @Timed
+    @Decompress
+    @Path("batchUpdate")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON_WITH_CHARSET)
+    public String update(@Context HugeConfig config,
+                         @Context GraphManager manager,
+                         @PathParam("graph") String graph,
+                         @QueryParam("updateStrategies")
+                         String updateStrategies,
+                         @QueryParam("check_vertex")
+                         @DefaultValue("true") boolean checkVertex,
+                         @QueryParam("create")
+                         @DefaultValue("true") boolean createIfNotExist,
+                         List<JsonEdge> jsonEdges) {
+        LOG.debug("Graph [{}] update edges: {}", graph, jsonEdges);
+        checkUpdatingBody(jsonEdges);
+
+        HugeGraph g = graph(manager, graph);
+        checkBatchSize(config, jsonEdges);
+
+        Map<String, JsonEdge> maps = new HashMap<>(jsonEdges.size());
+        Map<String, Object> strategies = parseProperties(updateStrategies);
```

Comment:
Define a class wrap all QueryParams and jsonEdges, then deserialize UpdateStrategies will be easy. you can refer 
`CustomizedPathsAPI.PathRequest`